Use opt for legacy PM for LLVM14.0+

burntfalafel opened the pull request:
Hi @sampsyo, this is a PR for users to use the legacy PM for [LLVM14.0 onwards](https://github.com/AFLplusplus/AFLplusplus/issues/1100). It adds the following things

- Use `opt` for running the pass on LLVM bitcode, as the latest `clang` versions use the new PM by default.
- Pass name through `skeleton` flag name to `opt`

It might be best to put this in a new branch if new users use your tutorials as a guide for newer LLVM versions. Thanks for making the lectures public - have helped me greatly :)!

sampsyo commented:
Awesome; thanks for looking into this! Updating the skeleton for current LLVM would be great.

We can switch to using `opt` if we absolutely have to, but for convenience's sake, I'm really interested in figuring out whether we can preserve the interaction where it only takes a single invocation of `clang` to run both standard passes and the custom plugin pass. (FWIW, doing this was _never_ recommended in the LLVM docs, even in the legacy PM era; the closest thing to a canonical source for how to do this was [my blog post](https://www.cs.cornell.edu/~asampson/blog/clangpass.html).)

I notice that your updated code still uses `RegisterStandardPasses`. I am a little confused, because I think that method _should_ enable a pass so it runs automatically when you run `clang`:
https://github.com/llvm/llvm-project/blob/56ad791f462fc4fc37213696f29b8c0a5e9404f8/llvm/include/llvm/Transforms/IPO/PassManagerBuilder.h#L241

Is that not the case? That is, does `RegisterStandardPasses` not do what I assume it does? (And if it doesn't, I think a workflow that relies on explicit `opt` invocation should _not_ use that, right?)

burntfalafel commented:
sampsyo, I reverted the code back and made the clang a one liner call. Only a simple explicit flag to invoke the legacy pass manager had to be called. You are right, the `RegisterStandardPasses` should enable the pass when running `clang` automatically. 

One thing I also added. the 'llvm::BasicBlock()` now has it's own `print()` function (no idea why :P). It expects the second argument to be `null`, thus removing the need for a `bool` value which is needed by `llvm::Value::print()`.

sampsyo commented:
Got it; that'll do! Thank you very much for your help! I'll update the various branches herein.
